Source/Scene/Material.js

```diff
@@ -878,13 +888,12 @@ function createTexture2DUpdateFunction(uniformId) {
           : Resource.createIfNeeded(uniformValue);
 
         var promise;
-        if (ktxRegex.test(resource.url)) {
-          promise = loadKTX(resource);
-        } else if (crnRegex.test(resource.url)) {
-          promise = loadCRN(resource);
+        if (ktx2Regex.test(resource.url)) {
+          promise = loadKTX2(resource.url);
         } else {
           promise = resource.fetchImage();
         }
+
         when(promise, function (image) {
```

Comment:
I think this code was here already, but we should never call `when(promise` like this, we should always use `promise.then`.  Also, it appears that there is no error handling here, which is weird.  Can you write up a clean-up issue for us to review promises in Material.js

Reply:
opened https://github.com/CesiumGS/cesium/issues/9610